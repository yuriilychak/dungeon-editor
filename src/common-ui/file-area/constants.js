
const stylePrefix = "file-area-";

export const EMPTY_VALUE = { id: -1, title: "Default" };
export const TREE_DATA = [{}];
export const STYLES = {
    ROOT: `${stylePrefix}root`,
    TITLE: `${stylePrefix}title`,
    DROP_AREA: `${stylePrefix}drop-area`,
    CLOSE: `${stylePrefix}close`,
    CLOSE_DISABLED: `${stylePrefix}close-disabled`
};
