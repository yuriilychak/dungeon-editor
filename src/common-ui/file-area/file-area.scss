@import "../../sizes";
@import "../../colors";
@import "../../other";

.file-area {
  $field-size: $item-size_small - $indent_s;

  &-root {
    @extend %flex-row;
    font-size: $font-size_medium;
    height: $item-size_small;
    min-width: $item-size_big;
    background-color: $back-deep-dark;
    border: $border-main;
    padding-left: $indent_m;
    flex: 1;
    box-sizing: border-box;
    overflow: hidden;
    position: relative;
    user-select: none;
  }

  &-title {
    flex: 1;
  }

  &-close {
    @extend %flex-row;
    background-color: $back-main;
    cursor: pointer;
    width: $field-size;
    margin-right: $indent_xs;
    font-size: $font-size_medium;
    height: $item-size_micro;
    justify-content: center;
    user-select: none;

    &-disabled {
      opacity: 0.5;
      pointer-events: none;
    }
  }

  &-drop-area {
    position: absolute;
    left: 0;
    right: $item-size_small;
    top: 0;
    bottom: 0;
  }

  :global {
    .rstcustom__ {
      &node {
        max-height: $item-size_small;
      }
    }
  }
}
