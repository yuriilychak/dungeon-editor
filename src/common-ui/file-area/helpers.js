export const canDrag = () => false;
export const canDrop = ({node}) => !node.isDirectory;
export const generateNodeProps = node => node;
export const cloneValue = data => ({ id: data.id, title: data.title });
