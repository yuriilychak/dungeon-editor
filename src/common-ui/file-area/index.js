export { default as FileArea } from "./file-area";
