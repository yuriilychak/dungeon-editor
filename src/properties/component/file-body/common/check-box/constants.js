const stylePrefix = "check-box-";

export const STYLES = {
    ROOT: `${stylePrefix}wrapper`,
    LABEL: {
        label: `${stylePrefix}label-label`,
        root: `${stylePrefix}label-root`
    },
    CHECK_BOX: {
        root: `${stylePrefix}icon-root`
    }

};
