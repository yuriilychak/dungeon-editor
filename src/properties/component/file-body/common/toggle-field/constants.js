import VisibilityIcon from '@material-ui/icons/Visibility';
import PanToolIcon from '@material-ui/icons/PanTool';
import FormatSizeIcon from '@material-ui/icons/FormatSize';
import SwapVertIcon from '@material-ui/icons/SwapVert';
import SwapHorizIcon from '@material-ui/icons/SwapHoriz';

import {STAGE_ELEMENT_PROP} from "../../../../../enum";

const stylePrefix = "properties-toggle-field";

export const ICONS = {
    [STAGE_ELEMENT_PROP.VISIBLE]: VisibilityIcon,
    [STAGE_ELEMENT_PROP.INTERACTIVE]: PanToolIcon,
    [STAGE_ELEMENT_PROP.TEXT_AUTO_SIZE]: FormatSizeIcon,
    [STAGE_ELEMENT_PROP.STRETCH_WIDTH]: SwapHorizIcon,
    [STAGE_ELEMENT_PROP.STRETCH_HEIGHT]: SwapVertIcon
};

export const STYLES = {
    ROOT: `${stylePrefix}-root`,
    BUTTON: `${stylePrefix}-button`,
    BUTTON_UNCHECKED: `${stylePrefix}-button-unchecked`
};
