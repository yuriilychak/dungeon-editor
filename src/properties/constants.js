export const VALUE_FORMAT = {
    PIXEL: "PX",
    PERCENT: "%",
    DEGREE: "Â°",
    COLOR: "CL",
    BOOL: "BL",
    POINT: "PT",
    TEXT: "TX"
};

export const FIELD_TYPE = {
    POINT: "point",
    COLOR: "color",
    SLIDER: "slider",
    CHECKBOX: "checkbox",
    NUMBER: "number",
    TOGGLE_GROUP: "toggleGroup",
    ENABLED: "enabled",
    TEXT: "text"
};

export const PROPERTY_SECTION = {
    COMMON: "common",
    TEXT: "text",
    NONE: ""
};
