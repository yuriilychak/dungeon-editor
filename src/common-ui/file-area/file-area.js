import React, {useCallback, useMemo, memo} from "react";
import classNames from "classnames";
import SortableTree from "react-sortable-tree";
import FileExplorerTheme from "react-sortable-tree-theme-file-explorer";
import {string, shape, number, func} from "prop-types";

import { EMPTY_VALUE, TREE_DATA, STYLES } from "./constants";
import { canDrag, canDrop, generateNodeProps, cloneValue } from "./helpers";

import "./file-area.scss";

export const FileArea = ({
                             dropId,
                             label,
                             value,
                             onChange
                         }) => {
    const isEmpty = value.id === EMPTY_VALUE.id;
    const handleChangeWithClone = useCallback(data => onChange(cloneValue(data)), [onChange]);
    const closeStyle = useMemo(() => classNames(STYLES.CLOSE, { [STYLES.CLOSE_DISABLED]: isEmpty }), [isEmpty]);
    const handleChange = useCallback(data => handleChangeWithClone(data[0]), [handleChangeWithClone]);
    const handleClose = useCallback(() => handleChangeWithClone(EMPTY_VALUE), [handleChangeWithClone]);

    return (
        <div className={STYLES.ROOT}>
            <span className={STYLES.TITLE}>
            {value.title}
            </span>
            <span
                className={closeStyle}
                onClick={handleClose}
            >
                âœ•
            </span>
            <div className={STYLES.DROP_AREA}>
                <SortableTree
                    maxDepth={0}
                    dndType={dropId}
                    treeData={TREE_DATA}
                    onChange={handleChange}
                    theme={FileExplorerTheme}
                    canDrag={canDrag}
                    isVirtualized={false}
                    canDrop={canDrop}
                    scaffoldBlockPxWidth={0}
                    generateNodeProps={generateNodeProps}
                    shouldCopyOnOutsideDrop
                />
            </div>
        </div>
    );
};

FileArea.propTypes = {
    label: string.isRequired,
    value: shape({
        id: number.isRequired,
        title: string.isRequired
    }).isRequired,
    dropId: string.isRequired,
    onChange: func.isRequired
};

export default memo(FileArea);
