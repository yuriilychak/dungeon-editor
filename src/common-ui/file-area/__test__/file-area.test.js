import React from "react";
import { createMount } from "@material-ui/core/test-utils";

import { FileArea } from "../file-area";
import { EMPTY_VALUE } from "../constants";

jest.mock("react-sortable-tree", () => ({
    onChange,
    canDrag,
    canDrop,
    generateNodeProps
}) => (
    <button
        id="SortableTree"
        onClick={() => onChange([{
            id: 2,
            title: "testId2"
        }])}
    >
        {`Sortable tree mock drag:${
            canDrag()} drop:${
            canDrop({ node: { isDirectory: false } })} nodeProps:${
            generateNodeProps("{ testProps }")
        }`}
    </button>
));

describe("GIVEN file-area", () => {
    const value = {
        id: 1,
        title: "testTitle1"
    };

    const onChange = jest.fn();
    const dropId = "testDropId";

    const wrapper = createMount()(
        <FileArea
            value={value}
            onChange={onChange}
            dropId={dropId}
        />
    );

    it("WHEN render component WITH default props THEN result should match snapshot", () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it("WHEN handle click on close element THEN onChange should call with empty object", () => {
        const element = wrapper.find("span").at(1);
        element.simulate("click");
        expect(onChange).toHaveBeenCalledWith(EMPTY_VALUE);
    });

    it("WHEN handle change THEN onChange should call with expected object", () => {
        const element = wrapper.find("button").at(0);
        element.simulate("click");
        expect(onChange).toHaveBeenCalledWith({
            id: 2,
            title: "testId2"
        });
    });
});