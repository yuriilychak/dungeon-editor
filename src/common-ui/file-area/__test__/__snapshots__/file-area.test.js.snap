// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIVEN file-area WHEN render component WITH default props THEN result should match snapshot 1`] = `"<div class=\\"file-area-root\\"><span class=\\"file-area-title\\">testTitle1</span><span class=\\"file-area-close\\">âœ•</span><div class=\\"file-area-drop-area\\"><button id=\\"SortableTree\\">Sortable tree mock drag:false drop:true nodeProps:{ testProps }</button></div></div>"`;
